.github/workflows/tests.yml

```diff
@@ -15,26 +15,25 @@ jobs:
           - macos-latest
         python-version:
           - '3.8'
+          
     runs-on: ${{ matrix.os }}
-    name: Run pytest
+    name: Check the installation builds
     steps:
-      - name: Checkout
-        uses: actions/checkout@v2
-
       - name: Set up Python
         uses: actions/setup-python@v2
         with:
           python-version: ${{ matrix.python-version }}
-
-      - name: Install local package
+      
+      - name: Install
         run: |
-          pip install wheel
-          pip install .
+          pip install --user wheel
+          pip install --user https://github.com/nestauk/grjobs/archive/${{ github.head_ref }}.zip
```

Comment:
there are issues with pip installing xgboost for macs (xgboost.core.XGBoostError) - I'd take it out of requirements.txt and allow the user to conda install it.